src/ILLink.RoslynAnalyzer/RequiresUnreferencedCodeAnalyzer.cs

```diff
@@ -85,11 +85,33 @@ public override void Initialize (AnalysisContext context)
 							IsNamedType (attrClass, "System.Diagnostics.CodeAnalysis.RequiresUnreferencedCodeAttribute") &&
 							attr.ConstructorArguments.Length == 1 &&
 							attr.ConstructorArguments[0] is { Type: { SpecialType: SpecialType.System_String } } ctorArg) {
+
+							StringBuilder methodName = new StringBuilder ();
+							if (method.MethodKind == MethodKind.PropertyGet || method.MethodKind == MethodKind.PropertySet) {
+								methodName.Append (method.ContainingType + "." + method.Name + "(");
+								bool shouldPrependComma = false;
```

Comment:
I assume this is to match the reporting done by the linker?
If that is the case we might want to consider slightly different solution - either change linker to report property getters/setters similarly to Roslyn, or adapt in the test verification.
Roslyn in general has better warning messages and better syntax to refer to code - "better" means, users I familiar with how Roslyn does it. We've already made changes in the linker to adapt to C#/Roslyn like type formatting and such, if there are gaps in that, we should change linker some more.

Reply:
Seems reasonable -- I didn't suggest this because I didn't know how comfortable we were with changing tests.

In that case the standard C# format is different in two ways here:

1. C# formats property accessors by saying the name of the property and then the name of the accessor, so instead of `get_MyProperty()` you'd have `MyProperty.set`.
2. C# uses type aliases by default instead of system type names, so `System.Int32` is `int`.

This is the default behavior of the C# analyzer framework -- VB has VB-specific formatting. It's up to you if you want to standardize on C# formatting, or try to stick to IL formatting. I would agree that C# is probably the most common format users see.

Reply:
@mateoatr worked on making the linker formatting "better" so he will know the details of what is already done. I don't have a problem with either approach honestly, I just don't think that the Roslyn analyzer should adapt to linker. We should either keep them different (and compensate in tests) or adapt the linker to Roslyn.

There were discussions what is better for linker especially around generics, since technically linker can't reliably provide the same output as Roslyn (unresolved generic type references don't have enough info in IL to print out the same string). But in the end I think we still keep the C#-like formatting since that's what most people are used to. The alternative was to use the Reflection syntax, but that is sometimes rather weird.

As for the two examples above, I personally don't like the Roslyn way of describing properties (it doesn't match the actual code syntax in any way), but either format would work I think. Just as mentioned above, Roslyn analyzer should be consistent with other Roslyn analyzers, linker can be somewhat different. And for using the type aliases, I think @mateoatr did add a couple of them, so we can either extend that list, or again, keep linker different.

Reply:
IIRC we don't do any type aliasing, but I think it should be fairly simple to create such a list and use it in `GetDisplayName`. Before having the `GetDisplayName` extension for member references, we used to report the source location of a warning by simply using the `FullName` property of Cecil, which was pretty verbose (the tests [here](https://github.com/mono/linker/blob/master/test/Mono.Linker.Tests/Tests/GetDisplayNameTests.cs) give a good idea of how the linker formats different member signatures.)